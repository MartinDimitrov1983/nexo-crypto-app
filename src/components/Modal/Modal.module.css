.modalOverlay {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: rgba(0, 0, 0, 0.5);
    display: flex;
    justify-content: center;
    align-items: center;
    z-index: 200;
}

.modal {
    display: flex;
    width: 400px;
    min-height: 250px;
    background: white;
    padding: 20px;
    border-radius: 8px;
    max-height: 450px;
}

.modalContent {
    display: flex;
    flex: 1 1;
    flex-direction: column;
}

.modalBody {
    display: flex;
    flex-direction: column;
    align-items: center;
    overflow-x: auto;
    padding: 10px 0;
    position: relative;
    min-height: 220px;
}

.loading {
    min-height: 330px;
}
.modalHeader {
    color: #5b6064;
    font-weight: 800;
    font-size: 24px;
    text-transform: capitalize;
    margin: 0;
}

.modalFooter {
    display: flex;
    justify-content: center;
    margin-top: 16px;
}
.close-button {
    position: absolute;
    top: 10px;
    right: 10px;
    cursor: pointer;
    font-size: 20px;
}
