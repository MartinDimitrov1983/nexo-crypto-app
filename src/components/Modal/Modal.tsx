import React, { ReactNode, MouseEvent } from 'react';
import ReactDOM from 'react-dom';
import styles from './Modal.module.css';
import { Button, Loading } from '../';
import { CLOSE, MODAL_ROOT } from '../../constants';

interface ModalProps {
    isOpen: boolean;
    onClose: () => void;
    children: ReactNode;
    header: string;
    isLoading: boolean;
}

const Modal = ({
    isOpen,
    onClose,
    header,
    isLoading,
    children,
}: ModalProps) => {
    const handleOverlayClick = (event: MouseEvent<HTMLDivElement>) => {
        const target = event.target as HTMLDivElement;
        if (target.classList.contains(styles.modalOverlay)) {
            onClose();
        }
    };

    return isOpen
        ? ReactDOM.createPortal(
              <div className={styles.modalOverlay} onClick={handleOverlayClick}>
                  <div className={styles.modal}>
                      <div className={styles.modalContent}>
                          <p className={styles.modalHeader}>{header}</p>
                          <div className={styles.modalBody}>
                              {isLoading && <Loading />}
                              {children}
                          </div>
                          <div className={styles.modalFooter}>
                              <Button onClick={onClose}>{CLOSE}</Button>
                          </div>
                      </div>
                  </div>
              </div>,
              document.getElementById(MODAL_ROOT) as HTMLElement,
          )
        : null;
};

export default Modal;
