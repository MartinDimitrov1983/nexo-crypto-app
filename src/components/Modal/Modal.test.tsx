import React from 'react';
import { render } from '@testing-library/react';

import Modal from './Modal';

test('Snapshot Modal', () => {
    const { asFragment } = render(
        <Modal
            isOpen={false}
            onClose={() => {}}
            header="test"
            isLoading={true}
            children={null}
        />,
    );
    expect(asFragment()).toMatchSnapshot();
});
