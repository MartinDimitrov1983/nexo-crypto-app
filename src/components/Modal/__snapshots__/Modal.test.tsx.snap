// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Modal 1`] = `<DocumentFragment />`;
